lib/Sema/CSApply.cpp

```diff
@@ -7957,6 +7961,9 @@ bool ExprRewriter::requiresDistributedThunk(Expr *base, SourceLoc memberLoc,
     [&](ParamDecl *P) {
     return P->isIsolated() ||
            llvm::is_contained(solution.isolatedParams, P);
+  },
+    [&](ParamDecl *P) {
+    return P->isDistributedKnownToBeLocal(); // TODO: need the solution too?
```

Comment:
Yeah, you'll want to have the solution contain a set of known-to-be-local-parameters, like we did for isolated parameters, and do the `llvm::is_contained` check there.